lib/botRepository.js

```diff
@@ -30,11 +30,12 @@ class BotRepository {
   }
 
   getLastPolledDate() {
-    return Rx.Observable.fromPromise(this.redis.get(LAST_POLLED_DATE));
+    return Rx.Observable.fromPromise(this.redis.get(LAST_POLLED_DATE))
+      .select(date => new Date(parseInt(date, 10)));
   }
 
   setLastPolledDate() {
-    this.redis.set(LAST_POLLED_DATE, new Date());
+    this.redis.set(LAST_POLLED_DATE, (new Date()).getTime());
```

Comment:
how about `new Date().toJSON()`? that would let you construct the date back from the string.